recipes/libcurl/all/conanfile.py

```diff
@@ -262,25 +262,20 @@ def _patch_mingw_files(self):
                 tools.save(os.path.join('lib', 'Makefile.am'), added_content, append=True)
 
     def _build_with_autotools(self):
-        env_run = RunEnvironment(self)
-        # run configure with *LD_LIBRARY_PATH env vars
-        # it allows to pick up shared openssl
-        self.output.info("Run vars: " + repr(env_run.vars))
-        with tools.environment_append(env_run.vars):
```

Comment:
By removing `RunEnvironment`, does this recipe still build with shared `ssh2`/`openssl`/`nghttp2`/`zlib`?

Reply:
it is still present, just using a newer syntax `self.run('./buildconf', win_bash=use_win_bash, run_environment=True)`

Reply:
No, `./buildconf` runs `autorconf` (which regenerates the `configure` script).
The `run_environment` is required in `autotools.configure()`.

Reply:
Finally had the chance to test it locally `conan create . -o openssl:shared=True` passes and prints the correct Openssl version

Reply:
Could you please shared the output?

Reply:
https://gist.github.com/prince-chrismc/e1247a4f3c13dcecc88ca59e0f656769

https://github.com/openssl/openssl/releases/tag/OpenSSL_1_1_1 version I have installed, which is not the one reported by the logs

From my understanding this is was we are expecting

Reply:
> No, `./buildconf` runs `autoreconf` (which regenerates the `configure` script).
> The `run_environment` is required in `autotools.configure()`.

Turns out this was true for clang =)